Furball.Engine/Engine/DevConsole/ImGuiConsole.cs

```diff
@@ -0,0 +1,141 @@
+                Hoverable   = true,
+                Depth       = 0
+            };
+
+            FurballGame.DrawableManager.Add(_consoleInputCoverDrawable);
+
+            style.ScrollbarRounding = 0;
+            style.ChildRounding     = 0;
+            style.FrameRounding     = 0;
+            style.PopupRounding     = 0;
+            style.WindowRounding    = 0;
+
+            style.Colors[(int) ImGuiCol.WindowBg]         = new Vector4((40.0f / 255.0f), (5.0f / 255.0f),  (50.0f / 255.0f), 0.8f);
+            style.Colors[(int) ImGuiCol.FrameBg]          = new Vector4((50.0f / 255.0f), (10.0f / 255.0f), (90.0f / 255.0f), 0.8f);
+            style.Colors[(int) ImGuiCol.ChildBg]          = new Vector4((50.0f / 255.0f), (10.0f / 255.0f), (90.0f / 255.0f), 0.8f);
+            style.Colors[(int) ImGuiCol.TitleBgActive]    = new Vector4((70.0f / 255.0f), (10.0f / 255.0f), (90.0f / 255.0f), 0.8f);
+            style.Colors[(int) ImGuiCol.TitleBgCollapsed] = new Vector4((70.0f / 255.0f), (10.0f / 255.0f), (90.0f / 255.0f), 0.8f);
+
+            style.WindowPadding = new Vector2(8, 8);
+            style.ItemSpacing   = new Vector2(8, 8);
+            style.FramePadding  = new Vector2(8, 4);
+        }
+
+        public static unsafe void Draw() {
+            if (Visible) {
+                ImGui.SetNextWindowSize(new Vector2(520, 600), ImGuiCond.FirstUseEver);
+                ImGui.Begin("Console", ref Visible, ImGuiWindowFlags.NoScrollbar | ImGuiWindowFlags.NoCollapse);
+
+                _consoleInputCoverDrawable.Position     = ImGui.GetWindowPos() / FurballGame.VerticalRatio;
+                _consoleInputCoverDrawable.OverrideSize = ImGui.GetWindowSize() / FurballGame.VerticalRatio;
+
+                float heightReserved = ImGui.GetStyle().ItemSpacing.Y + ImGui.GetFrameHeightWithSpacing();
+
+                ImGui.BeginChild("ScrollingRegion", new Vector2(0,           -heightReserved), true, ImGuiWindowFlags.HorizontalScrollbar);
+                ImGui.PushStyleVar(ImGuiStyleVar.ItemSpacing, new Vector2(4, 1));
+
+                for (int i = 0; i != _devConsoleCache.Count; i++) {
+                    ConsoleResult result = _devConsoleCache[i];
+
+                    switch (result.Result) {
+                        case ExecutionResult.Success:
+                        case ExecutionResult.Log:
+                        case ExecutionResult.Message:
+                            ImGui.PushStyleColor(ImGuiCol.Text, new Vector4(255, 255, 255, 255));
+                            break;
+                        case ExecutionResult.Warning:
+                            ImGui.PushStyleColor(ImGuiCol.Text, new Vector4(255, 255, 0, 255));
+                            break;
+                        case ExecutionResult.Error:
+                            ImGui.PushStyleColor(ImGuiCol.Text, new Vector4(255, 0, 0, 255));
+                            break;
+                    }
+
+                    ImGui.SetScrollY(ImGui.GetScrollMaxY());
```

Comment:
You are setting the scroll position to the bottom every frame, instead of when something new is added, which means that scrolling is always forced to the bottom

Reply:
right, this should only happen when u enter something